project_timesheet_time_control/models/account_analytic_line.py

```diff
@@ -25,9 +25,14 @@ class AccountAnalyticLine(models.Model):
     @api.model
     def _eval_date(self, vals):
         if vals.get('date_time'):
-            return dict(vals, date=fields.Date.to_date(vals['date_time']))
+            return dict(vals, date=self._convert_datetime_to_date(vals['date_time']))
         return vals
 
+    def _convert_datetime_to_date(self, datetime_):
+        if isinstance(datetime_, str):
+            datetime_ = fields.Datetime.from_string(datetime_)
+        return fields.Date.context_today(self, datetime_)
+
```

Comment:
This should be enough, no need to add an extra method:

```suggestion
            return dict(
                vals,
                date=fields.Date.to_date(fields.Datetime.to_datetime(vals['date_time']))
            )
        return vals
```

Reply:
@Yajo I disagree about your suggestion. First, it breaks the behavior. Second, the extra method makes the intension more explicit.

Reply:
What does it break?

Reply:
@Yajo it does not use the timezone of the user. Your solution is similar to the original behavior before this pull request.

Reply:
OK, how about this?

```suggestion
            ts = fields.Datetime.to_datetime(vals['date_time'])
            ts = fields.Date.context_today(ts)
            return dict(vals, date=ts)
```

Reply:
@Yajo ok, but there's a small error in your proposal:

ts = fields.Date.context_today(ts)

should be

ts = fields.Date.context_today(self, ts)